### PR TITLE
Anticheat bypass fix

### LINKED ISSUE #2124: Have you been banned? Ban data collection thread
### Issue at hand

YimMenu appears to be collecting quite a few bans lately, I'm using this issue to collect data on what's causing the bans so we can figure out a root cause.

### Anyone coming here please answer the following questions:

```
Date of ban:
YYYY/MM/DD

How often do you play GTA V :
Daily

How many hours do you spend weekly on GTA V :
1 hour

Did you play any missions : 
Yes/No

Did you use any Lua's? : 
Yes/No

If yes, which Lua's did you use? : 

Did you ever cheat in any money? :
Yes/No

How much did you cheat in approx. :
1 million

Did you load any other menu's next to YimMenu? : 
...

Which commit hash is your version of YimMenu.dll (grab it from the latest cout.log)? :


```

### PR DESCRIPTION
**Pull request by maybegreat48**
Fixes #2124 

Requires testing

### CODE REVIEW

**Comment by gir489returns** [comment]
This might be a problem to test. Since I wasn't banned before, having me test this now, is just going to show I either still won't get banned, or got banned and don't know if it was just delayed or this caused it. If they were did a ban wave before and just chose to sit on the info now, we could have a false positive that we've gone back to being FUD until the next wave. I guess, the only way to know for sure that we're not detected is if we can debug the metrics we're sending and see if we send any that are out of bounds.

**Comment by gir489returns** [comment]
[cout.log](https://github.com/YimMenu/YimMenu/files/12613851/cout.log)

image


I just tried it, and it crashed as soon as I went into the game.

**Comment by gir489returns** [comment]
Still crashing with the 2nd commit, just not as frequently.

image

**Comment by gir489returns** [comment]
image

This PR is still failing due to a crash caused by clicking Kill All Enemies under World. Cause: unknown.

**Comment by Sys0xdbg** [comment]
its leaked publicly why not add it. 
` 
main.add("AXINTEGRITY", "48 8D 45 ? 48 89 45 ? 48 8D 05 ? ? ? ? 48 89 45", [&](ptr_manage h) {
        void* targetaddress = h.add(8).as<void*>();
        patch(targetaddress, 0x90, 7);
});
`
i take no credits. originally made by Mr Monarch
EinTim
#makeyimmenugreatagain

File: src/byte_patch_manager.cpp [hunk 1]
```diff
@@ -83,6 +83,9 @@ namespace big
 
 		// Always send the special ability event
 		memory::byte_patch::make(g_pointers->m_gta.m_activate_special_ability_patch, std::to_array({0xB0, 0x01, 0xC3}))->apply();
+
+		// Crash the game if the game manages to send a report
+		memory::byte_patch::make((PVOID)g_pointers->m_gta.m_bonus_transaction, std::vector(0x60, 0x0))->apply();
```
Comment (Yimura) [inline src/byte_patch_manager.cpp]: Couldn't we hook this to send a log at the least so we know this is happening or are you relying on the stack trace to convey this?

Edit: didn't mean to approve

**Review by Yimura — DISMISSED** [review]

**Comment by gir489returns** [comment]
> its leaked publicly why not add it.
> ` main.add("AXINTEGRITY", "48 8D 45 ? 48 89 45 ? 48 8D 05 ? ? ? ? 48 89 45", [&](ptr_manage h) { void* targetaddress = h.add(8).as<void*>(); patch(targetaddress, 0x90, 7); });`
> i take no credits. originally made by Mr Monarch
> EinTim
> #makeyimmenugreatagain

We don’t hook anything that’s integrity checked.

File: src/byte_patch_manager.cpp (hunk 1 above)
Comment (gir489returns) [reply src/byte_patch_manager.cpp:88]: It’s just a JMP to some encrypted code. I guess you could resolve the JMP and then inject there, call print stack trace, then kill the app.

**Comment by TwystedMind** [comment]
Sorry to jump in here, this was a thought that I had.  Is there any suspicion that the newer "Relay IP Connection" feature may be setting off flags with Rockstar?  Or are you thinking that isn't the case?  That feature wasn't added too long ago, and Rockstar is known for their delayed suspensions/bans when flagged by the Anti-Cheat.  I know it's a feature myself and many other use as part of the protections, and it is very network focused.

Just a thought on the subject.  Sorry if this is out of place.

**Comment by maybegreat48** [comment]
Thanks to @gir489returns for testing. The sigscanner reports appear to now be properly blocked

**Comment by ghost** [comment]
> Sorry to jump in here, this was a thought that I had. Is there any suspicion that the newer "Relay IP Connection" feature may be setting off flags with Rockstar? Or are you thinking that isn't the case? That feature wasn't added too long ago, and Rockstar is known for their delayed suspensions/bans when flagged by the Anti-Cheat. I know it's a feature myself and many other use as part of the protections, and it is very network focused.
> 
> Just a thought on the subject. Sorry if this is out of place.

It's probably not an issue with regard to anti-cheat. I've been experimenting with relay connections with firewall rules and through the yimmenu protection setting since the feature was released and haven't experienced anything except broken mission progression as a result. Handy for hiding one's ip address but only really useful outside of missions, contracts, and heists.

**Comment by TwystedMind** [comment]
> > Sorry to jump in here, this was a thought that I had. Is there any suspicion that the newer "Relay IP Connection" feature may be setting off flags with Rockstar? Or are you thinking that isn't the case? That feature wasn't added too long ago, and Rockstar is known for their delayed suspensions/bans when flagged by the Anti-Cheat. I know it's a feature myself and many other use as part of the protections, and it is very network focused.
> > Just a thought on the subject. Sorry if this is out of place.
> 
> It's probably not an issue with regard to anti-cheat. I've been experimenting with relay connections with firewall rules and through the yimmenu protection setting since the feature was released and haven't experienced anything except broken mission progression as a result. Handy for hiding one's ip address but only really useful outside of missions, contracts, and heists.

I was wondering because when using a relay connection, our IPs are going to change frequently, if not when joining a new session, at least with every re-launch.  I'm not sure if Rockstar cares enough about that, but seeing one Social Club account/MAC Address running from various locations across the country/world on a daily basis may look suspicious, no?

Again, I don't really know if Rockstar cares enough to pay attention to that, they don't IP ban (at least not yet), because that's easy to get around.  It was just a thought, but I see your side of the point as well.

Thank you for responding!

**Comment by ghost** [comment]
> > > Sorry to jump in here, this was a thought that I had. Is there any suspicion that the newer "Relay IP Connection" feature may be setting off flags with Rockstar? Or are you thinking that isn't the case? That feature wasn't added too long ago, and Rockstar is known for their delayed suspensions/bans when flagged by the Anti-Cheat. I know it's a feature myself and many other use as part of the protections, and it is very network focused.
> > > Just a thought on the subject. Sorry if this is out of place.
> > 
> > 
> > It's probably not an issue with regard to anti-cheat. I've been experimenting with relay connections with firewall rules and through the yimmenu protection setting since the feature was released and haven't experienced anything except broken mission progression as a result. Handy for hiding one's ip address but only really useful outside of missions, contracts, and heists.
> 
> I was wondering because when using a relay connection, our IPs are going to change frequently, if not when joining a new session, at least with every re-launch. I'm not sure if Rockstar cares enough about that, but seeing one Social Club account/MAC Address running from various locations across the country/world on a daily basis may look suspicious, no?
> 
> Again, I don't really know if Rockstar cares enough to pay attention to that, they don't IP ban (at least not yet), because that's easy to get around. It was just a thought, but I see your side of the point as well.
> 
> Thank you for responding!

I know this isn't the place for this but just a clarification your ip does not change using the feature. It just routes/tunnels your traffic through rockstars matchmaking servers. This masks your ip from other players. It does not mask it from rockstar. So instead of a peer-to-peer connection you'd have a peer-to-rockstar-to-peer connection. Hope that helps clear things up.

**Review by Yimura — APPROVED** [review]

**Comment by Yimura** [comment]
After this is merged we can't guarantee that the anti-cheat is fully defeated again and that YimMenu is safe to use again. To anyone using this we'll only know it's safe after at the least 30 days of usage.

**Pull request merged**